### DISCUSSION TITLE
Feature Request: Hierarchical Issues in Projects

### DISCUSSION THREAD

**Discussion by Soyokaze-42** [community/community#5714] [Projects and Issues] [answered]
I would love to see a system where I can have an issue be a part of a larger issue in support of feature planning. I use another platform for agile planning, and this would bring GitHub one step closer to working for me. I have Features that are composed of many Stories, but I have not found a way to associate issues with a larger artifact.

I'm using SAFe where there are 4 defined layers of artifacts Epics -> Capabilities -> Features -> Stories. If issues could have a type field and we could make a one-to-many relationship between issues, that would be a start.

When viewing an issue with relationships, it would be nice to be able to view/manage linked issues.

**Comment by atorres-eqrx** [comment]
I'm strongly in favor of this request. For now I'm using a workaround https://github.com/marketplace/actions/epic-issues-for-github which does work, but is a bit unintuitive because you have to bounce around different screens to first create the issue, the the issue number, then find the epic, and then update the description. Having first-class support for this hierarchy would be a huge improvement.

**Comment by mversluys** [comment]
A related suggestion was for there to be a field type which is an issue https://github.com/github/feedback/discussions/6421 which would allow for hierarchies to be expressed as long as the thing being referred to is an issue.

**Reply by white-sdev** [reply]
Please! make them with infinite sublevels, It is very anoying to find yourself limited by just 1 level, this will be deal braker for me :(
[#9913](https://github.com/github/feedback/discussions/9913), [#7813](https://github.com/github/feedback/discussions/7813)

**Comment by zonofthor** [comment]
This is a very frequent request and has been asked of in many different threads here, using various terminology. Unfortunately this spreads out thin the demand for this feature. (and I have no solution to offer)

**Comment by bsutton** [comment]
Just adding my voice.
It was the first issue I hit. I need to be able to nest tasks to at least three of four levels deep.

**Comment by feerrenrut** [comment]
This seems to be described on the public roadmap, but without a specific timeline (only 'future'): https://github.com/github/roadmap/issues/284

**Reply by clshaw01** [reply]
So they ditched that effort and now it's this as of Feb 2024: https://github.com/github/roadmap/issues/927

**Comment by sgbaird** [comment]
Does anyone have experience using GitHub + Jira integration? https://support.atlassian.com/jira-cloud-administration/docs/integrate-with-github/

**Comment by rileybroughten** [comment, marked as answer]
Hello! 👋 Tasklists add support for issue hierarchy at GitHub and are currently in Private Beta! You can request access and join the waitlist [here](https://github.com/features/issues/signup).

For more information on how these work, check out the [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-tasklists).

**Reply by Soyokaze-42** [reply]
Thanks! I have been approved for the beta to evaluate this solution.
